### ISSUE TITLE
Add fee and total breakdown on payments

### ISSUE THREAD

**Issue by arrenv** [JoinColony/colonyCDapp#2869]
### Story

As a User, I want to be aware of and see a full break down of all fees for a payment.

### Description

When payments are made that leave a colony, they are subject to the Colony network fee. This fee is a vital component of Colony's business model, allowing the network to operate and be maintained. While the network fee is currently as low as possible at 1 wei of the token being used for payment, this will not always be the case. Nonetheless, it is important to be transparent about these fees and provide this information clearly and prominently in the interface.

This enhancement involves adding a new element to all payment actions (i.e., Simple and all Advanced) that displays a fee and total breakdown in a prominent position both when creating and reviewing the payment.

### Design

[Figma link](https://www.figma.com/design/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?node-id=26076-279778&t=Cjerr218hYFP944L-4)

![image](https://github.com/user-attachments/assets/06ae564d-d7a1-4527-852c-3c13c6a07bfc)

![image](https://github.com/user-attachments/assets/7ddd3eb4-d178-4b61-abe8-ef580397e51d)

## Acceptance criteria
- **General**
    - Should be in alignment with desktop/tablet/mobile design from Designs.
    - Should use existing components and styles where possible for consistency.
    - Should use skeleton loaders.
    - Should use tooltips as of design.s

- **Network fee draw**
    - Fee draw should sit over the action content.
    - User can click the whole bar to open and close the draw. Draw should also close when clicking out of the draw.
    - The total and fee values should update as the amount field is updated, by keystroke.
    - The max height of the fee draw should be 50% of the action container.
        - If the contents exceed this height, it should vertically scroll, with content aligned to the bottom, with scroll up.
    - It should show the percent of the network fee.
        - The fee percent value should use the live network fee during creation or during the motion, then it should use the payments stored network fee after successful payment.
        - If the network fee is below 0.0001%, then the percent value should not be shown.
    - Collapsed draw
        - Total value
            - The summation of all tokens in the payment converted to the users nominated currency including fees.
                - If an individual value has no value and is thus converted to 0, it will count as 0 in the total.
                - Such that, if all values have 0 value when converted, then value will show as 0, regardless of the amount of tokens.
            - The total should be shown in the users nominated currency.
                - If some or all of the value is not able to be calculated e.g. has no price, we should show a warning icon as of the design.
    - Expanded draw
        - Token breakdown
            - Fees and breakdown should be separated by token type.
            - We should show the total for the token in the tokens currency.
            - The token converted value should be shown in the users nominated currency.
                - If the token has no price, we should show the warning icon and the message “Price not available”.
            - Fee breakdown value
                - The fee value should be in the same currency as the token symbol, which, will be the same as the amount field.
                - With low network fee values, we should show the abbreviated decimal, with subscript for the number of decimal digits. e.g. $0.0_{17}1$ CLNY

**Comment by jakubcolony** [comment]
@arrenv does this issue also include displaying all the fees associated with crypto off-ramp payments? If so I will provide some guidelines on how and when to calculate those

**Comment by arrenv** [comment]
@jakubcolony Thank yoU!
Yes, it will need to include those once we have KYB implemented.
As well as fees for performing bridging, swaps and other custom actions that include fees.

**Comment by jakubcolony** [comment]
There are 2 types of fees we're dealing with:

### Colony Network Fee

- Applies to all payments that aren't crypto off-ramp (see below)
- There's a hook for retrieving it (`useNetworkInverseFee`) and helpers to calculate final amount (`getAmountWithFee`, `getAmountLessFee`)
- To get the percentage, use the following formula: `1/fee * 100%`

### Crypto Off-ramp Fees
Crypto off-ramp is any payment meeting these conditions:

- Token is USDC
- Recipient has `isAutoOfframpEnabled` flag enabled in their profile
- Recipient has a valid liquidation address

You can use [`adjustRecipientAddress`](https://github.com/JoinColony/colonyCDapp/blob/dc22d82118f1fa26a8ea64f53240742ac467fb0c/src/redux/sagas/utils/expenditures.ts#L239) as a reference. It would be great to extract that logic into a shared helper to avoid duplication.

The crypto off-ramp fee has 2 components:

- Gateway fee, a percentage returned from the backend
- Transfer fee, a fixed fee which depends on the recipient's chosen payment currency/rail

The logic for calculating these is largely implemented in `useTransferFees`. However, currently the crypto to fiat integration doesn't support getting the payment rail of another user.

@arrenv just to clarify - does this mean there will be no relevant fees shown for crypto off-ramp payments to KYC-verified recipients until KYB is implemented? If so, then we only need to check if a payment is not crypto off-ramp (and calculate network fee). Otherwise, we should prioritize #4203 which will also make the calculations possible.

**Comment by arrenv** [comment]
@jakubcolony Yes, correct. The fees for KYC are covered on the recipient side and not by the colony, the idea is that a colony is not aware that a user is converting to fiat on their side, so showing the fees in this step does not quite seem accurate.

This is at least the thought process for now.
